### PR TITLE
Allow laravel to end workers with lost connection

### PR DESCRIPTION
**Pull request by violarium**
Exception thrown by ::pop is checked by Illuminate\Database\DetectsLostConnections::causedByLostConnection
It should throw exception with special message in order to end queue on error correctly.
Otherwise it continues to work with broken connection.

https://github.com/vyuldashev/laravel-queue-rabbitmq/issues/421
https://github.com/vyuldashev/laravel-queue-rabbitmq/issues/436

### CODE REVIEW

**Comment by roboticsexpert** [comment]
it fixes the issue !

**Comment by Shipu** [comment]
It's solve the issue but still sometime lost connection.

**Comment by violarium** [comment]
> It's solve the issue but still sometime lost connection.

This fix is not to prevent lost connection - it can happen because of network or something else.

When connection is lost Laravel worker should restart queue. But with rabbitmq it doesn't do it. It just continueы to work with lost connection, spamming errors.

This fix makes Laravel work with queue properly and restart it.

**Comment by amiranbarigreenweb** [comment]
This is what im looking for ...

**Review by Shipu — APPROVED** [review]

**Review by amiranbarigreenweb — APPROVED** [review]

**Comment by ar2r** [comment]
Did this code helps someone? Now i inject this code in my project and after 2-3 days i will tell my result.

**Comment by violarium** [comment]
> Did this code helps someone? Now i inject this code in my project and after 2-3 days i will tell my result.

I'am an author of this. You better not inject, but use this: https://github.com/vyuldashev/laravel-queue-rabbitmq/issues/436#issuecomment-1000133104

For my projects it works fine for 2 months. It, ofcource, doesn't prevent lost connection to rabbit. But worker get restarted by laravel as it was intended.

**Comment by kovalovme** [comment]
@vyuldashev please review

**Comment by xanloboi** [comment]
@vyuldashev , please review

**Comment by henrylynch** [comment]
Morning! Do anyone know when this will be reviewed?

**Review by vyuldashev — APPROVED** [review]

**Comment by vyuldashev** [comment]
Sorry, I am a little bit busy at the moment. Thanks for the contribution. Released under v12.0.1

**Comment by kovalovme** [comment]
Thanks a lot!

**Comment by kovalovme** [comment]
Any chance you can support 11.x version?
Laravel 9 migration is pretty complex

Thank you again!

File: src/Queue/RabbitMQQueue.php [hunk 1]
```diff
@@ -254,6 +257,16 @@ public function pop($queue = null)
             }
 
             throw $exception;
+        } catch (AMQPChannelClosedException|AMQPConnectionClosedException $exception) {
```
Comment (BinZhiZhu) [inline src/Queue/RabbitMQQueue.php:260]: 👍🏻

**Review by BinZhiZhu — COMMENTED** [review]
👍🏻

**Pull request merged**